src/main/java/Duke.java

```diff
@@ -1,10 +1,365 @@
+
+            boolean isListModified = false;
+
+            // Prints the list of tasks stored if "list" is called
+            if (commandModified.equals("list")) {
+                printList(listOfTasks);
+            } else if (commandModified.contains("done")) {
+                // Update done status for indicated task
+                doneTask(command, listOfTasks);
+                isListModified = true;
+            } else if (commandModified.startsWith("delete")) {
+                // Delete the indicated task
+                deleteTask(listOfTasks, command);
+                isListModified = true;
+            } else if (!saidBye) {
+                // Store the command into the array as a task if it's none of the above
+                try {
+                    addTask(command, listOfTasks, taskCount);
+                    isListModified = true;
+                } catch (InvalidCommandException e) {
+                    // Informs user when command is inserted without stating the type of task
+                    botSpeak("☹ Sorry but I don't understand that at all. Try again?");
+                }
+            }
+
+            // Update the txt file if something is added/deleted/marked as done
+            if (isListModified) {
+                try {
+                    DukeFiles.writeToFile(listOfTasks);
+                } catch (IOException e) {
+                    System.out.println("There's a problem with writing the file");
+                }
+            }
+        } while (!saidBye);
+    }
+
+    /**
+     * Identifies the type of task given by user and add into the list
+     *
+     * @param command     user input at terminal
+     * @param listOfTasks ArrayList containing tasks inserted by user
+     * @param taskCount   Store the amount of tasks inserted
+     * @throws InvalidCommandException exception due to commands without specifying the type
+     */
+    private static void addTask(String command, ArrayList<Task> listOfTasks, int taskCount) throws InvalidCommandException {
+        String task;
+
+        // Identifies the task type
+        TaskType taskType = getTaskType(command);
+
+        // Creates new object based on the type of the task
+        switch (taskType) {
+        case TODO:
+            // Extract the string after "todo"
+            task = command.trim().substring("todo".length()).trim();
+            try {
+                listOfTasks.add(new ToDo(task));
+                listOfTasks.get(taskCount).printAddResult();
+            } catch (InvalidCommandException e) {
+                botSpeak("☹ OH NO! The description of todo cannot be empty!");
+            }
+            break;
+        case DEADLINE:
+            // Command inserted by user will be processed and added into the list of tasks
+            try {
+                listOfTasks.add(new Deadline(command));
+                listOfTasks.get(taskCount).printAddResult();
+            } catch (InvalidCommandException e) {
```

Comment:
Nice to see different levels of catches (InvalidCommandException, InvalidDateException...). Good job!